apis/Google.Cloud.Storage.V1/Google.Cloud.Storage.V1/RetryHandler.cs

```diff
@@ -37,46 +37,64 @@ internal sealed class RetryHandler : IHttpUnsuccessfulResponseHandler
         internal const string InvocationIdHeaderPart = "gccl-invocation-id";
         internal const string AttemptCountHeaderPart = "gccl-attempt-count";
 
-        private static readonly int[] s_retriableErrorCodes =
-        {
-                408, // Request timeout
-                429, // Too many requests
-                500, // Internal server error
-                502, // Bad gateway
-                503, // Service unavailable
-                504 // Gateway timeout
-        };
-        private static readonly RetryHandler s_instance = new RetryHandler();
+        private double _backoffMultiplier;
+        private TimeSpan _initialBackoff;
+        private TimeSpan _maxBackoff;
+        private RetryPredicate _retryPredicate;
 
         private RetryHandler() { }
 
+        //This method to be completely replaced by the following overloaded method once the implementation across the library is complete
         internal static void MarkAsRetriable<TResponse>(StorageBaseServiceRequest<TResponse> request)
         {
+            request.AddExecuteInterceptor(InvocationIdInterceptor.Instance);
+        }
+
+        internal static void MarkAsRetriable<TResponse>(StorageBaseServiceRequest<TResponse> request, RetryOptions options)
+        {
+            RetryHandler retryHandler = new RetryHandler()
```

Comment:
Possibly create a private constructor that takes  `RetryOptions` and use it to initialize the handler.